linkml_runtime/utils/schemaview.py

```diff
@@ -738,11 +738,21 @@ def is_multivalued(self, slot_name: SlotDefinition) -> bool:
         """
         returns True if slot is multivalued, else returns False
         :param slot_name: slot to test for multivalued
-        :return boolean: 
+        :return boolean:
         """
         induced_slot = self.induced_slot(slot_name)
         return True if induced_slot.multivalued else False
 
+    @lru_cache()
+    def slot_has_quality(self, slot_name: SlotDefinition, quality_to_check) -> bool:
```

Comment:
missing type annotation and param in docstring?

Not sure about naming. This is hard without getting all meta, but perhaps unavoidable. I'd rather avoid introducing new terms like "quality"

- slot_has_metaslot
- slot_has_slot_assignment (we use "slot assignment" in the spec)

I also recommend addition an optional class name argument - if this is present then class induced slot is used

Reply:
ah I see this is intended for boolean only - maybe have more defensive checking

or is the intent this for both 

1. slots with range boolean
2. slots that are not required, checking for whether the optional value is set
3. checking for empty strings and integers != 0

That feels like overloading. I would stick to 1

for checking for booleans, perhaps for a name

- slot_is_true_for?
- slot_has_property
- ...?

Reply:
with the changes - do you still think I should handle class_name argument?